forum/models/base_model.py

```diff
@@ -0,0 +1,50 @@
+"""
+Database models for forum.
+"""
+
+from abc import ABC, abstractmethod
+
+from forum.mongo import MongoBackend
+
+
+class MongoBaseModel(ABC):
+    """Abstract Class for Mongo model implementation"""
+
+    def __init__(self, collection_name, client=None):
+        self.client = client or MongoBackend(collection=collection_name)
+        self.collection = self.client.collection
+
+    @property
+    def get_collection(self):
+        """Get mongo db collection"""
+        return self.collection
+
+    @property
+    def get_client(self):
+        """Get mongo client"""
+        return self.client
+
+    @abstractmethod
+    def get(self, **kwargs):
+        """Get a document by filter"""
+        raise NotImplementedError
+
+    @abstractmethod
+    def list(self, **kwargs):
```

Comment:
We can also add default implementation.

self.get_collection.find(kwargs)

Reply:
updated